pkg/peerprotocol/connection.go

```diff
@@ -79,6 +89,23 @@ func NewConnection(ip *net.IP, options ...ConnectionOptionFunc) (*Connection, er
 	return c, nil
 }
 
+// NewServerConnection creates a new connection object with the specified peer
+func NewServerConnection(conn *websocket.Conn, options ...ConnectionOptionFunc) (*Connection, error) {
```

Comment:
Just curious when the case is we already have a websocket connection vs an IP to connect to?

Is there a reason we can't just use the `peerPort` for this? I don't really see a difference between serverPort and peerPort (and they default to the same config value unless overridden with config option)